lib/state/wallet/logic.dart

```diff
@@ -1793,4 +1794,54 @@ class WalletLogic extends WidgetsBindingObserver {
       launchUrl(Uri.parse(uri), mode: LaunchMode.externalApplication);
     } catch (_) {}
   }
+
+  void setWalletActionsLoading(bool loading) {
+    _state.setWalletActionsLoading(loading);
+  }
+
+  Future<void> evaluateWalletActions() async {
+    _state.setWalletActionsLoading(true);
+
+    _state.walletActions = [];
+
+    List<ActionButton> actionsToAdd = [];
+
+    final showVouchers = selectShowVouchers(_state);
```

Comment:
vouchers can always be shown. The underlying conditions that are being checked are not really important.